plugin/trino-hive/src/test/java/io/trino/plugin/hive/crypto/TestHiveParquetEncryption.java

```diff
@@ -0,0 +1,345 @@
+        ColumnEncryptionProperties ageProperties = ColumnEncryptionProperties.builder(AGE_PATH).withKey(COLUMN_KEY_AGE).build();
+
+        FileEncryptionProperties encodingProperties = FileEncryptionProperties.builder(FOOTER_KEY)
+                .withAlgorithm(ParquetCipher.AES_GCM_CTR_V1)
+                .withEncryptedColumns(Map.of(AGE_PATH, ageProperties, ID_PATH, idProperties))
+                .build();
+
+        try (ParquetWriter<Group> writer = ExampleParquetWriter.builder(new org.apache.hadoop.fs.Path(path.toString()))
+                .withType(schema)
+                .withConf(new Configuration())
+                .withEncryption(encodingProperties)
+                .withWriteMode(OVERWRITE)
+                .withPageSize(1024) // small pages -> dictionary likely
+                .build()) {
+            SimpleGroupFactory factory = new SimpleGroupFactory(schema);
+            for (int i = 0; i < 5000; i++) {
+                int id = i % 11;
+                if (id == missingId) {
+                    id = (id + 1) % 11;  // skip one value for id
+                }
+                int age = i % 11;
+                if (age == missingAge) {
+                    age = (age + 1) % 11;  // skip one value for age
+                }
+                writer.write(factory.newGroup().append("id", id).append("age", age));
+            }
+        }
+
+        // Verify both columns actually have dictionary pages
+        try (ParquetDataSource source = new FileParquetDataSource(path.toFile(), ParquetReaderOptions.defaultOptions())) {
+            FileDecryptionProperties dec = FileDecryptionProperties.builder()
+                    .withKeyRetriever(new TestHiveParquetEncryption.TestingParquetEncryptionModule(
+                            FOOTER_KEY, Optional.of(COLUMN_KEY_AGE), Optional.of(COLUMN_KEY_ID)))
+                    .build();
+            ParquetMetadata metadata = MetadataReader.readFooter(source, Optional.empty(), Optional.empty(), Optional.of(dec));
+
+            ColumnChunkMetadata ageChunk = metadata.getBlocks().getFirst().columns().stream()
+                    .filter(column -> column.getPath().equals(AGE_PATH))
+                    .findFirst().orElseThrow();
+            ColumnChunkMetadata idChunk = metadata.getBlocks().getFirst().columns().stream()
+                    .filter(column -> column.getPath().equals(ID_PATH))
+                    .findFirst().orElseThrow();
+
+            assertThat(ageChunk.getDictionaryPageOffset()).isGreaterThan(0);
+            assertThat(idChunk.getDictionaryPageOffset()).isGreaterThan(0);
+            assertThat(ageChunk.getEncodings()).anyMatch(org.apache.parquet.column.Encoding::usesDictionary);
+            assertThat(idChunk.getEncodings()).anyMatch(org.apache.parquet.column.Encoding::usesDictionary);
+        }
+    }
+
+    public static class TestingParquetEncryptionModule
+            implements Module, DecryptionKeyRetriever
```

Comment:
Instead of injecting Module into query runner, can we just use `io.trino.parquet.crypto.TestParquetEncryption.TestingKeyRetriever` and bind that in `io.trino.plugin.hive.TestingHiveConnectorFactory` ?
Allowing arbitrary Module into query runner can in future be abused for complicated test setups